### ISSUE TITLE
File references lost in production

### ISSUE THREAD

**Issue by E2design** [nystudio107/craft#69]
### Describe the bug
Since the latest releases of the Vite build  & the Vite plugin the site in production mode seems to have lost connection to the build files. The files are build and correctly moved into the css/dist directory. But while opening the page in production (useDevServer' => false) there are no file references in the HEAD to these files.

I've installed a new version nystudio/craft vite set-up in a fresh new docker. And even with that setup the same issue occurs.

### To reproduce
Steps to reproduce the behaviour:
1. Change vite.php to production (useDevServer' => false)
2. Make build
3. Files are build see [this](https://cln.sh/wi5qiW) screenshot
4. Refresh web page

### Versions
- Plugin version: Vite plugin 1.0.17
- Craft version: 3.7.17.2

**Comment by khalwat** [comment]
Have you looked at the generated `manifest.json` to see if all of the expected files are there?

**Comment by E2design** [comment]
Yepp all [files](https://cln.sh/g3vLf0) are there.

**Comment by khalwat** [comment]
Okay, I know what this is... 

The build setup that used to work in Vite pre 2.6, but now appears to be broken. In the Vite root dir, I have an alias `src/` that aliases to a directory outside of the Vite root.  This is done for organizational purposes, when I build with `manifest: true` to output a `manifest.json`, the keys in the manifest used to look like this:

```
  "src/js/app.ts": {
```
but now it resolves the alias as part of the path:
```
  "../src/web/assets/src/js/app.ts": {
```
 ...which is not the desired result. Changing `resolve.preserveSymlinks` doesn't seem to have any effect.

I'll be creating a minimum reproducible repository, and filing an issue on the Vite issues page shortly.

**Comment by khalwat** [comment]
Filed an issue: https://github.com/vitejs/vite/issues/5415

**Comment by E2design** [comment]
Would there be any temporary work around? Facing a live production soon.

**Comment by khalwat** [comment]
Yeah you can just move the `src/` dir to where the alias is, and it should fix everything for now

I've considered adding a patch fix to the Vite plugin to work around the situation too, because I do want to keep the alias/file structure in place

**Comment by PjGeeroms** [comment]
For me personally on build it did not generate the complete path, it instead generated a key like this
`../src/js/app.js`. Changing the path to the script accordingly fixed the issue for me. Maybe because I created the alias with `ln -nfs` that it is actually following the alias now that makes key different.

(Using Vite 2.6.14)

**Comment by khalwat** [comment]
They've apparently addressed this in Vite 2.7.0-beta.4 -- I plan to have a look at it shortly.

**Comment by khalwat** [comment]
Confirmed fixed in `^2.7.0-beta.10` -- https://github.com/vitejs/vite/issues/5415#issuecomment-986429103
